src/components/ProblemsTable.tsx

```diff
@@ -68,58 +79,124 @@ export const ProblemsTable = ({
     [userFileTypeIcons]
   );
 
-  const expandReports = (reports: Report[]): { category: string; subRows: Report[] }[] => {
-    const expandableReports: Record<string, Report[]> = {};
+  const getFileNameFromId = React.useCallback(
+    (id?: string) => {
+      const filesInfo = sourceFilesInfo || context?.reportData.sourceFilesInfo;
+      return filesInfo?.fileId === id
+        ? filesInfo?.fileName
+        : filesInfo?.Files?.find((file) => file.fileId === id)?.fileName;
+    },
+    [sourceFilesInfo, context?.reportData.sourceFilesInfo]
+  );
+
+  const fileData = React.useMemo(() => {
+    const filesInfo = sourceFilesInfo || context?.reportData.sourceFilesInfo;
+    return [
+      { ...filesInfo, mainFile: true },
+      ...(filesInfo && filesInfo.Files ? filesInfo.Files.filter((file: SourceFile) => file.state !== 'Hidden') : []),
+    ];
+  }, [sourceFilesInfo, context?.reportData.sourceFilesInfo]);
+
+  const fileDataHash = React.useMemo(() => {
+    return fileData.reduce((previousValue, currentValue) => {
+      return { ...previousValue, [currentValue.fileId as string]: currentValue };
+    }, {}) as { [fileId: string]: SourceFile };
+  }, [fileData]);
+
+  const expandReports = React.useCallback(
+    (reports: Report[]) => {
+      const currentTable = context?.currentTable;
+      const expandableColumn = currentTable ? (tableStyleAccessor[currentTable] as keyof Report) : undefined;
+      const isFileTable = expandableColumn === 'fileId';
 
-    reports.forEach((report) => {
-      const category = report?.category;
+      if (!expandableColumn || context?.currentTable === 'problems') {
+        return reports;
+      }
+
+      const expandableReports: Record<string, Report[]> = {};
 
-      if (!category) {
-        if (!Object.hasOwn(expandableReports, 'Others')) {
-          expandableReports.Others = [];
+      reports.forEach((report) => {
+        const topLevel = report[expandableColumn];
+        if (!topLevel) {
+          if (!Object.hasOwn(expandableReports, 'Others')) {
+            expandableReports.Others = [];
```

Comment:
from the 'Report' Type, all fields except 'fileId' are optional. This was added in case a scenario occurs where a report doesn't have the property matching the expandable level (eg - no category), it will create a top level 'Others' for those reports to fall under